### PR TITLE
[Macros] Fix memory leak in `expandMacroExpr`

### PR DESCRIPTION
**Pull request by rxwei**
The result of `swift_ASTGen_lookupMacro` needs to be deallocated.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please smoke test

File: test/Macros/macro_plugin_exec.swift [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
 // RUN: %target-build-swift -L%platform-module-dir/../.. -enable-experimental-feature Macros -load-plugin-library %t/%target-library-name(MacroDefinition) %s -o %t/main
-// RUN: %target-run %t/main
+// RUN: %target-run %t/main | %FileCheck %s
 // REQUIRES: executable_test
 
 // FIXME: Swift parser is not enabled on Linux CI yet.
 // REQUIRES: OS=macosx
 
 print(#customStringify(3 + 2 - 1))
+// CHECK: (4, "3 + 2 - 1")
```
Comment (DougGregor) [inline test/Macros/macro_plugin_exec.swift:11]: Heh, I had these locally, too. Thanks!

File: lib/Sema/TypeCheckMacros.cpp [hunk 2]
```diff
@@ -159,6 +161,7 @@ Expr *swift::expandMacroExpr(
       return nullptr;
     evaluatedSource = NullTerminatedStringRef(evaluatedSourceAddress,
                                               (size_t)evaluatedSourceLength);
+    free(macro);
```
Comment (DougGregor) [inline lib/Sema/TypeCheckMacros.cpp:164]: Please use `swift_ASTGen_destroyMacro`

Comment (rxwei) [reply lib/Sema/TypeCheckMacros.cpp]: fixed

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

**Comment by rxwei** [comment]
@swift-ci please smoke test macOS and merge

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

**Pull request merged**